### DISCUSSION TITLE
FR | Order Settings | Disable order confirmation email

### DISCUSSION THREAD

**Discussion by manfield** [PrestaShop/PrestaShop#40204] [Ideas]
**Is your feature request related to a problem? Please describe.**
In some circumstances, especially when PS is integrated with a BackOffice system, we're in the need to disable Order Confirmation email. 
That's because the order confirmation is processed by the backoffice system and the user would receive more than one mail for his order.
I've seen that is hard coded without any check on any constant related to a possible configuration.

**Describe the solution you'd like**
Would be nice to have an additional toggle under Order Settings in order to disable confirmation emails for new valid orders.

**Describe alternatives you've considered**
I did not.

**Comment by khouloudbelguith** [comment]
Hi @manfield,

It's possible to activate or deactivate the mails to send in Shop parameters> Orders> orders statuses.
Thanks for check and feedback.

**Comment by manfield** [comment]
Actually not for the new order.
For any order history change that happens after the new order that's true, but not for the first new order event.

**Comment by khouloudbelguith** [comment]
@marionf, @colinegin what do you think?
Thanks!

**Comment by marionf** [comment]
@manfield you can delete the order_conf.html and order_conf.txt templates in the email/_lang_ folder

**Comment by manfield** [comment]
Sure it is a workaround, though many people - not so skilled - won't find this solution easily.
Why don't you add a toggle in configuration? I think it is a cleaner way.

**Comment by marionf** [comment]
@manfield 

We have posted your suggestion here: http://feedback.prestashop.com/forums/387864-prestashop-1-7-x/suggestions/37448746-disable-emails

Don't hesitate to vote for it

**Comment by Payalba** [comment]
Actually, i wrote a module and register the hook actionEmailSendBefore.

`$this->registerHook('actionEmailSendBefore')`

and do something like that:

```
 public function hookActionEmailSendBefore($params){

       if($params['template'] === 'order_conf'){
            return false;
        }
        return true;
    }

```
